src/key_info_managers/mod.rs

```diff
@@ -24,18 +27,21 @@ pub mod on_disk_manager;
 /// This structure corresponds to a unique identifier of the key. It is used internally by the Key
 /// ID manager to refer to a key.
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct KeyTriple {
-    app_name: ApplicationName,
-    provider_id: ProviderId,
+pub struct KeyIdentity {
+    /// The identity of the application that created the key.
+    application: ApplicationIdentity,
+    /// The identity of the provider where the key is stored.
+    provider: ProviderIdentity,
```

Comment:
What about the idea of having this as an `Option<ProviderIdentity>`, only required for the OnDisk KIM?

Reply:
Yeah maybe. I'll have a look into it more seeing what issues I run into implementing the `SQLiteKeyInfoManager` itself. E.g. what information I'm missing etc